lambdas/issuecredential/src/main/java/uk/gov/di/ipv/cri/kbv/api/service/EvidenceFactory.java

```diff
@@ -72,11 +72,23 @@ private CheckDetail[] createCheckDetailsWithKbvQuality(KBVItem kbvItem)
         var questionState = objectMapper.readValue(kbvItem.getQuestionState(), QuestionState.class);
 
         if (hasPassedWithOneIncorrectAnswer(kbvItem)) {
+            // Scenario:
+            // 1 question was wrong in the first batch
+            // And therefore 2 questions were given in second batch
+            // And they were answered correctly
+            // So we can sort by quality and remove neccessary question
             if (questionState.allQuestionBatchSizesMatch(2)) {
                 return createCheckDetailsBySortingOnKbvQuality(kbvItem, questionState);
             }
+            // Scenario:
+            // 2 questions in first batch were correct
+            // 3rd question in the second batch was incorrect
+            // And 4th question in the third batch was correct
+            // So we can remove the third question
```

Comment:
Yes, we skip the 3rd question, because we know precisely which which 3 to map to KBVQuality